### ISSUE TITLE
Can't get source maps to work with --inspect

### ISSUE THREAD

**Issue by oliversturm** [anmonteiro/lumo#290]
Hi,

I'm building a test project to JavaScript using this:

```
(require '[lumo.build.api :as b])

(b/build "src"
         {:main 'test.core
          :output-dir "out"
          :output-to "out/main.js"
          :optimizations :none
          :target :nodejs})
```

The file `out/test/core.js` has some large blob of stuff appended that starts like this:

```
//# sourceURL=out/test/core.js?rel=1507209794443
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoib3V0L3Rl
... 30 or so more lines
```

What is that? Some encoded form of the source map? The weirdest URL in history? I'm not an expert on source maps, and I notice that for `out/cljs/core.js` there is a separate `core.js.map` - so something seems to be working here, but perhaps not for my own file?

In any case I now run `node --inspect-brk out/main.js`, and in Chrome I can bring up the debugger (`chrome://inspect`) and see my code. There is a big problem though: Chrome only shows the file `out/main.js` in the hierarchy. Since I'm working locally now, I can add my local source folder to the hierarchy (this makes no difference as far as I can see, but it's an option). However, I plan to run the code in a Docker image in the end, which means I'd be connecting remotely to the node process. I don't see how I could now access any of the other source code files, perhaps to set a break point.

Now, if I step through the code in `main.js`, I can see how Chrome adds files to the hierarchy. However, my own code file `out/test/core.js` or its corresponding `.cljs` equivalent never appear in the list. So I'm altogether at a loss... any help appreciated!

Oli

**Comment by anmonteiro** [comment]
The ClojureScript self-hosted compiler generates inline source-maps, which is what you're seeing at the end of files.

I suspect the reason you don't see source maps in `node --inspect` is because you haven't installed the `source-map-support` package?

This is documented in the ClojureScript quick-start (search for `source-map-support` [here](https://clojurescript.org/guides/quick-start))

**Comment by oliversturm** [comment]
Yes, you are correct - I forgot to install that package. My apologies :)

Incidentally, why is the start-up code so stupid:

```js
try {
    require("source-map-support").install();
} catch(err) {
}
```

If I configure my build to include source maps, wouldn't it be better to just leave out the try/catch, so I would actually see an error if I don't have the package? No idea where this code is generated...

I looked around a bit and I found that this seems to be a more general problem of Chrome. For instance, this issue: https://github.com/nodejs/node/issues/8369

I had more success debugging with VS Code - it can access the source map as intended, though it has similar issues providing the source initially to set breakpoints. I'll keep looking, but now I'm reasonably sure that the source map is available an intact, so I'll close this issue. Thanks for your help!
